src/ripple/basics/SlabAllocator.h

```diff
@@ -30,67 +30,177 @@
 
     static_assert(alignof(Type) == 8 || alignof(Type) == 4);
 
+    // A simple linked list node to keep track of the free list.
+    //
+    // The nodes for this linked list are constructed inside unused memory
+    // in the slab's allocated block.
+    //
+    // By way of example, on a system with 1 byte long pointers, where the
+    // slab allocator allocates 4 byte long items from a 16 byte long slab
+    // the memory layout is as follows:
+    //
+    //     slot: 0      1      2      3
+    //           [ .... | .... | .... | ....]
+    //
+    // We don't need anything too complicated: the only operations that we
+    // perform are insertion and removal, but only at the head of the list
+    // which means that all we need is very basic CAS.
+    //
+    // Normally, to keep a linked list we need a "node" structure, similar
+    // to this. And, if we're going to have a node, we need some memory in
+    // which to "store" it.
+    //
+    // But remember that we only need a "node" when an item is going to be
+    // on the free list, and something is on the free list because there's
+    // some memory that is otherwise unused.
+    //
+    // So if the size of the unused memory is greater than the size of our
+    // node (which is the size of a pointer) then we can use that directly
+    // and overlay a singly linked list of free objects onto the memory of
+    // those free objects.
+    //
+    // Here's what the memory may look like if slots 1 and 3 are allocated
+    // and slots 0 and 2 are free, with the free list overlaid on top:
+    //
+    //     list_ --------------+
+    //                         |
+    //                         v
+    //     slot: 0      1      2      3
+    //           [ N... | .... | N... | .... ]
+    //           ^ |             |
+    //           | '---> NULL    |
+    //           |               |
+    //           '---------------'
+    //
+    // Notice that although the first empty slot in memory order is slot 0
+    // the free list points to slot 2 as the first available slot and slot
+    // 2 points to slot 0, which points to NULL, as it is it the last free
+    // slot.
+    //
+    // Now imagine that the object in slot 1 is released. The free list is
+    // updated to point to slot 1 as the first available slot. The  memory
+    // now looks like this:
+    //
+    //  list_ ----------+
+    //                  |
+    //                  v
+    //     slot: 0      1      2      3
+    //           [ N... | N... | N... | .... ]
+    //           ^ |      |    ^ |
+    //           | |      |    | |
+    //           | |      '----' |
+    //           | '---> NULL    |
+    //           |               |
+    //           '---------------'
+    //
+    // Note that the order of slots within the free list at any given time
+    // is random, as it depends on the pattern and timing of memory access
+    // of the application.
+    struct alignas(Type) FreeItem
```

Comment:
I'm seeing a bunch of warnings of the form: 
```
src/ripple/basics/SlabAllocator.h:115:12: error: requested alignment is less than minimum alignment of 8 for type 'ripple::SlabAllocator<ripple::SHAMapItem>::FreeItem'
```

Reply:
I just repro'd these on macOS clang 15 as well.